scrapy/extensions/closespider.py

```diff
@@ -60,6 +64,19 @@ def __init__(self, crawler: Crawler):
             crawler.signals.connect(
                 self.item_scraped_no_item, signal=signals.item_scraped
             )
+        if self.close_on.get("pagecount_no_item"):
+            if self.page_count not in dispatcher.getReceivers(
+                signal=signals.response_received
+            ):
+                crawler.signals.connect(
+                    self.page_count, signal=signals.response_received
+                )
+
+            if self.item_scraped not in dispatcher.getReceivers(
+                signal=signals.item_scraped
+            ):
+                crawler.signals.connect(self.item_scraped, signal=signals.item_scraped)
```

Comment:
What if instead of using dispatcher you extend the `if` statement above?

e.g.

```python
        if self.close_on.get("pagecount") or self.close_on.get("pagecount_no_item"):
            crawler.signals.connect(self.page_count, signal=signals.response_received)
```